tests/salmon_tests/__init__.py

```diff
@@ -1,17 +1,24 @@
 import os
 import shutil
+import subprocess
+import time
 
 import config.testing  # noqa
 
 
 dirs = ["run", "logs"]
+DUMMY_RELAY=None
 
 
 def setup_package():
     for path in dirs:
         os.mkdir(path)
+    DUMMY_RELAY=subprocess.Popen("python -m smtpd -n -c DebuggingServer localhost:7899 > logs/smtpd.log", shell=True)
+    time.sleep(10)
```

Comment:
It all boils down to this comment on the original code:

https://github.com/moggers87/salmon/blob/master/tests/salmon_tests/server_tests.py#L126
"this test actually delivers to a test server"

My understanding of the lamson testing code is that it assumes you're running a stmp at 8899. By virtue of silently failing if there is none, all these tests seem to be correct, when they were not.

The relay was not being tested. It was failing and continuing silently.

My initial fix was to set a "in_test" argument in Relay to mimic that behavior, but that didn't work with command_tests.

How do you want to reuse the smtplib mock?